src/pages/api/newsletter/subscribe.ts

```diff
@@ -0,0 +1,41 @@
+import prisma from '@/lib/prisma';
+import { NextResponse } from 'next/server';
+
+export async function POST(request: Request) {
+  try {
+    const { email } = await request.json();
+
+    if (!email || !email.includes('@')) {
+      return NextResponse.json(
+        { error: 'Please provide a valid email address' },
+        { status: 400 },
+      );
+    }
+
```

Comment:
We may wanna check for duplicate emails:

const existingSubscriber = await prisma.contacts.findUnique(email);

    if (existingSubscriber) {
      return status 409 you are subscribed already
    }